### PR TITLE
Add email validity check

### PR DESCRIPTION
**Pull request by chrisshyi**
Added email validity checking for the following actions:
- User update
- User email addition
  - Through the web interface
  - Through the `/user/emails` endpoint
- User creation:
  - Through the admin interface
  - Through the `/admin/users` endpoint

### CODE REVIEW

File: models/user.go [hunk 1]
```diff
@@ -957,6 +963,26 @@ func updateUser(e Engine, u *User) error {
...
+		return err
+	}
+
+	u.Email = strings.ToLower(u.Email)
+	isExist, err := sess.
+		Where("email=?", u.Email).
+		Get(new(User))
+	if err != nil {
+		return err
+	} else if isExist {
```
Comment (lunny) [inline models/user.go]: But if the user is himself?

File: options/locale/locale_zh-TW.ini [hunk 2]
```diff
@@ -364,6 +364,7 @@ org_name_been_taken=組織名稱已被使用。
 team_name_been_taken=團隊名稱已被使用。
 team_no_units_error=請至少選擇一個儲存庫區域。
 email_been_used=此電子信箱已被使用
+email_invalid=無效的電子信箱
```
Comment (lunny) [inline options/locale/locale_zh-TW.ini]: Please don't submit the file

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/user.go (hunk 1 above)
Comment (chrisshyi) [reply models/user.go]: Yup, I didn't consider that scenario. Fixed!

File: options/locale/locale_zh-TW.ini (hunk 2 above)
Comment (chrisshyi) [reply options/locale/locale_zh-TW.ini]: Ok, fixed!

File: models/user.go [hunk 3]
```diff
@@ -957,6 +963,17 @@ func updateUser(e Engine, u *User) error {
 
 // UpdateUser updates user's information.
 func UpdateUser(u *User) error {
+	sess := x.NewSession()
```
Comment (lafriks) [inline models/user.go]: Session is not needed here

**Review by lafriks — APPROVED** [review]

**Review by lafriks — CHANGES_REQUESTED** [review]
Wrong button, see comment above

File: models/user.go (hunk 3 above)
Comment (chrisshyi) [reply models/user.go]: ok, fixed!

**Review by lafriks — APPROVED** [review]

File: models/user.go [hunk 4]
```diff
@@ -957,6 +963,11 @@ func updateUser(e Engine, u *User) error {
 
 // UpdateUser updates user's information.
 func UpdateUser(u *User) error {
+	u.Email = strings.ToLower(u.Email)
```
Comment (techknowlogick) [inline models/user.go:960]: could this logic be moved to `updateUser` so that all the update user logic is kept together.

**Review by techknowlogick — CHANGES_REQUESTED** [review]

File: models/user.go (hunk 4 above)
Comment (chrisshyi) [reply models/user.go:960]: ok, done!

**Review by 6543 — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
Thanks :)

**Comment by techknowlogick** [comment]
@lunny please review

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
-> #13666

**Pull request merged**